tests/rptest/tests/partition_reassignments_test.py

```diff
@@ -277,17 +314,54 @@ def test_reassignments_kafka_cli(self):
         reassignments_json = self.make_reassignments_for_cli(
             all_node_idx, initial_assignments)
 
-        kafka_tools = KafkaCliTools(self.redpanda)
-        output = kafka_tools.execute_reassign_partitions(
-            reassignments=reassignments_json)
+        output = self.execute_reassign_partitions(
+            reassignments=reassignments_json, timeout_s=30)
         check_execute_reassign_partitions(output, reassignments_json,
                                           self.logger)
 
-        output = kafka_tools.verify_reassign_partitions(
+        output = self.verify_reassign_partitions(
             reassignments=reassignments_json, timeout_s=180)
         check_verify_reassign_partitions(output, reassignments_json,
                                          self.logger)
 
+    def alter_partition_reassignments_with_kcl(
+            self,
+            topics: dict[str, dict[int, list[int]]],
+            user_cred: Optional[dict[str, str]] = None,
+            timeout_s: int = 10):
+        ok_re = re.compile(
```

Comment:
now there is some regex logic in the test and some in kcl for this request, IMO I would place all of the request parsing logic in KCL and add a retries argument to the method. There are other client requests that work like this (see rpk.describe_configs(retries=5), etc) and throw an exception if the number of retries has exceeded.

Then your test would just boil down to ensuring you've only recieved OK's as a response from KCL.

Reply:
@dotnwat raised a good point about checking `unknown_topic_or_partition` in the test rather than in the client wrapper.
https://github.com/redpanda-data/redpanda/pull/8849#discussion_r1113734364

I think the solution is to include `timeout_sec` param. In the future where a dev wants to check what happens when a topic is undefined, they can then adjust the timeout to their expectations.